nexus/src/external_api/http_entrypoints.rs

```diff
@@ -2065,19 +2095,69 @@ async fn instance_create(
     let new_instance_params = &new_instance.into_inner();
     let handler = async {
         let opctx = OpContext::for_external_api(&rqctx).await?;
-        let instance = nexus
-            .project_create_instance(
+        let project_id = nexus
+            .project_lookup_id(
                 &opctx,
-                &organization_name,
-                &project_name,
-                &new_instance_params,
+                params::ProjectSelector::ProjectAndOrg {
+                    project_name: project_name.clone().into(),
+                    organization_name: organization_name.clone().into(),
+                },
             )
             .await?;
+        let instance = nexus
+            .project_create_instance(&opctx, &project_id, &new_instance_params)
+            .await?;
         Ok(HttpResponseCreated(instance.into()))
     };
     apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
 }
 
+/// Path parameters for Instance requests
+#[derive(Deserialize, JsonSchema)]
+struct InstanceLookupPathParam {
+    /// If Name is used to reference the instance you must also include one of the following qualifiers as query parameters:
+    /// - `project_id`
+    /// - `project_name`, `organization_id`
+    /// - `project_name`, `organization_name`
+    ///
+    /// If Id is used the above qualifiers are will be ignored
+    instance: NameOrId,
+}
+
+#[derive(Deserialize, JsonSchema)]
+struct InstanceQueryParams {
+    #[serde(flatten)]
+    selector: params::ProjectSelector,
```

Comment:
I think the `flatten` here causes there to be duplicates in the output because the selector is an untagged enum. Need to figure out how to fix that.

Reply:
Actually, I believe I was incorrect. I think this is likely a schemars issue. I was able to work around it in [d0a19c6](https://github.com/oxidecomputer/omicron/pull/1957/commits/d0a19c6669e6427c24989a4d79fe84ac687a9bf4) by using `#[schemars(skip)]` for the duplicate names, but we should dig into where this is failing and make the relevant PR.